File: JobMatcher.Service/JobMatcher.Api/ViewModels/JobSeekerProfileViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;
using JobMatcher.Models;
using JobMatcher.Service.Mapping;

namespace JobMatcher.Service.ViewModels
{
    public class JobSeekerProfileViewModel : IMapFrom<JobSeekerProfile>, IHaveCustomMappings
    {
        [Key]
        public int JobSeekerProfileId { get; set; }

        public string Username { get; set; }

        public string Email { get; set; }

        public ProfileType ProfileType { get; set; }

        public string Summary { get; set; }

        public virtual Image Image { get; set; }

        public virtual ICollection<ExperienceViewModel> Experience { get; set; }

        public virtual ICollection<EducationViewModel> Education { get; set; }

        public virtual ICollection<ProjectViewModel> Projects { get; set; }

        public virtual ICollection<SkillViewModel> Skills { get; set; }

        public virtual ICollection<JobOfferViewModel> SelectedJobOffers { get; set; }

        public virtual ICollection<Match> Matches { get; set; }

        public virtual ICollection<MessageViewModel> Messages { get; set; } 

        public void CreateMappings(AutoMapper.IConfiguration configuration)
        {
            configuration.CreateMap<JobSeekerProfile, JobSeekerProfileViewModel>()
                .ForMember(m => m.Username, opt => opt.MapFrom(x => x.User.UserName))
                .ForMember(m => m.Email, opt => opt.MapFrom(x => x.User.Email))
                .ForMember(m => m.ProfileType, opt => opt.MapFrom(x => x.User.ProfileType))
                .ReverseMap();
        }
    }
}

File: JobMatcher.iOS/JobMatcher.iOS/Models/JobSeekerProfileViewModel.m
//
//  JobSeekerProfileViewModel.m
//  JobMatcher.iOS
//
//  Created by s i on 1/30/16.
//  Copyright © 2016 svetlai. All rights reserved.
//

#import "JobSeekerProfileViewModel.h"
#import "ProjectViewModel.h"
#import "SkillViewModel.h"
#import "EducationViewModel.h"
#import "ExperienceViewModel.h"
#import "JobOfferViewModel.h"
#import "MessageViewModel.h"

@implementation JobSeekerProfileViewModel

-(instancetype) initWithId:(NSInteger) profileId
                  andEmail:(NSString*) email
               andUsername:(NSString*) username
                andSummary:(NSString*) summary
               andProjects:(NSArray*) projects
                 andSkills:(NSArray*) skills
             andExperience:(NSArray*) experience
              andEducation:(NSArray*) education
      andSelectedJobOffers:(NSArray*) selectedJobOffers
               andMessages:(NSArray*) messages
            andProfileType:(NSInteger) profileType
              andFirstName:(NSString*) firstName
               andLastName:(NSString*) lastName
            andPhoneNumber:(NSString*) phoneNumber
        andCurrentPosition:(NSString*) currentPosition{
    
    if (self = [super init]){
        self.profileId = profileId;
        self.email = email;
        self.username = username;
        self.summary = summary;
        self.projects = projects;
        self.skills = skills;
        self.experience = experience;
        self.education = education;
        self.selectedJobOffers = selectedJobOffers;
        self.messages = messages;
        self.profileType = profileType;
        self.firstName = firstName;
        self.lastName = lastName;
        self.phoneNumber = phoneNumber;
        self.currentPosition = currentPosition;
    }
    
    return self;

}

+(JobSeekerProfileViewModel*) fromJsonDictionary:(NSDictionary *)jsonDictionary{
    return [[JobSeekerProfileViewModel alloc]
            initWithId:[[jsonDictionary objectForKey:@"JobSeekerProfileId"] integerValue]
            andEmail:[jsonDictionary objectForKey:@"Email"]
            andUsername:[jsonDictionary objectForKey:@"Username"]
            andSummary:[jsonDictionary objectForKey:@"Summary"]
            andProjects:[ProjectViewModel arrayOfProjectsFromJsonDictionary:[jsonDictionary objectForKey:@"Projects"]]
            andSkills:[SkillViewModel arrayOfSkillsFromJsonDictionary:[jsonDictionary objectForKey:@"Skills"]]
            andExperience:[ExperienceViewModel arrayOfExperienceFromJsonDictionary:[jsonDictionary  objectForKey:@"Experience"]]
            andEducation:[EducationViewModel arrayOfEducationFromJsonDictionary:[jsonDictionary objectForKey:@"Education"]]
            andSelectedJobOffers:[JobOfferViewModel arrayOfJobOffersFromJsonDictionary:[jsonDictionary  objectForKey:@"SelectedJobOffers"]]
            andMessages:[MessageViewModel arrayOfMessagesFromJsonDictionary:[jsonDictionary objectForKey:@"Messages"]]
            andProfileType:[[jsonDictionary objectForKey:@"ProfileType"] integerValue]
            andFirstName:[jsonDictionary objectForKey:@"FirstName"]
            andLastName:[jsonDictionary objectForKey:@"LastName"]
            andPhoneNumber:[jsonDictionary objectForKey:@"PhoneNumber"]
            andCurrentPosition:[jsonDictionary objectForKey:@"CurrentPosition"]];
            ;
}

+(NSArray*) arrayOfJobOffersFromJsonDictionary: (NSArray*) jsonArray{
    NSMutableArray *arr = [NSMutableArray array];
    for (NSInteger i = 0; i < jsonArray.count; i++) {
        [arr addObject:[JobSeekerProfileViewModel fromJsonDictionary:jsonArray[i]]];
    }
    
    return [NSArray arrayWithArray:arr];
}

@end
